src/wallet/rpcwallet.cpp

```diff
@@ -1731,23 +1755,55 @@ UniValue listsinceblock(const JSONRPCRequest& request)
         filter = filter | ISMINE_WATCH_ONLY;
     }
 
+    bool include_removed = (request.params.size() < 4 || request.params[3].get_bool());
+
     int depth = pindex ? (1 + chainActive.Height() - pindex->nHeight) : -1;
 
     UniValue transactions(UniValue::VARR);
+    std::set<uint256> listed;
 
     for (map<uint256, CWalletTx>::iterator it = pwalletMain->mapWallet.begin(); it != pwalletMain->mapWallet.end(); it++)
     {
         CWalletTx tx = (*it).second;
 
         if (depth == -1 || tx.GetDepthInMainChain() < depth)
-            ListTransactions(tx, "*", 0, true, transactions, filter);
+        {
+            if (ListTransactions(tx, "*", 0, true, transactions, filter) && include_removed)
+            {
+                listed.insert(it->first);
+            }
+        }
+    }
+
+    // when a reorg'd block is requested, we also list any relevant transactions
+    // in the blocks of the chain that was detached
+    UniValue removed(UniValue::VARR);
+    while (include_removed && paltindex && paltindex != pindex)
+    {
+        CBlock block;
+        if (!ReadBlockFromDisk(block, paltindex, Params().GetConsensus()))
+        {
+            throw JSONRPCError(RPC_INTERNAL_ERROR, "Can't read block from disk");
```

Comment:
Options:
①. Try to respect the path taken which requires the blocks as removed transactions do not track the block they were in, meaning if a node prunes out the block, the operation fails.
②. Ignore the path and simply show all removed transactions between the fork point and the chain tip, which requires NO loading of blocks (faster), but means the caller will get irrelevant removed transactions in reorgs it was never aware of to begin with.

Pros and cons with ①:
1. Pro: Caller gets exactly what they need to update their view of the chain state, and is not confused by irrelevant stuff.
2. Con: Slower due to block read requirement.
3. Con: Fails if blocks involved in the reorg were pruned out.

Pros and cons with ②:
1. Pro: Faster due to no block read.
2. Pro: Works fine even if involved blocks were pruned.
3. Con: Does not distinguish between the various forks involved.